### PR TITLE
CUI-7416 Coral-Spectrum Tooltip is not included in accessibility name when referenced using aria-labelledby

### PR DESCRIPTION
**Pull request by majornista**
## Description
In Chrome, foundation-autocomplete in AEM is not including the error tooltip text as part of the accessibility name even though the appropriate IDREF is included in the aria-labelledby attribute for the combobox input.

Possible solution is to set visibility: initial on the coral-tooltip animation to close has completed. Chrome seems to ignore the element with display: none, visibility: hidden, and aria-hidden="true", when calculating the accessibility name.

## Related Issue
[CUI-7416](https://jira.corp.adobe.com/browse/CUI-7416)

## Motivation and Context
For accessibility, when a form element is invalid, we need to communicate the reason it is invalid to assistive technology.

## How Has This Been Tested?
Updated Tooltip example documentation to set aria-describedby on target element for tooltip. Accessibility description updates when the target tooltip changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by majornista** [comment]
@mohiaror, @icaraps, @indra2gurjar

File: coral-base-overlay/src/scripts/BaseOverlay.js [hunk 1]
```diff
@@ -527,7 +527,12 @@ const BaseOverlay = (superClass) => class extends superClass {
       // Set aria-hidden false before we show
       // Otherwise, screen readers will not announce
       // Doesn't matter when we set aria-hidden true (nothing being announced)
-      this.setAttribute('aria-hidden', !open);
+      if (open) {
```
Comment (mohiaror) [inline coral-base-overlay/src/scripts/BaseOverlay.js:530]: You might want to change the comment above with respect to this change.

**Comment by mohiaror** [comment]
Looks good to me.

**Comment by mohiaror** [comment]
@icaraps , @indra2gurjar  can you please review and merge this PR? It is for a long pending sev2 accessibility issue which is being flagged.

File: coral-component-actionbar/src/tests/test.ActionBar.js [hunk 2]
```diff
@@ -652,19 +652,20 @@ describe('ActionBar', function() {
       bar.primary.items.add({}).appendChild(new Button());
      
       // Wait for resize listener
-      window.setTimeout(() => {
+      window.setTimeout(function() {
```
Comment (icaraps) [inline coral-component-actionbar/src/tests/test.ActionBar.js:655]: What's the reason for using `function` over the arrow function ?

Comment (majornista) [reply coral-component-actionbar/src/tests/test.ActionBar.js:655]: Arrow function is only used for these `setTimeout`s. With test being flaky, I was trying to be consistent.

**Comment by icaraps** [comment]
How does this visibility change affect the overlays like Popover or Dialog ?

**Comment by majornista** [comment]
> How does this visibility change affect the overlays like Popover or Dialog ?

@icaraps I see no noticeable change in the behavior. On hide, we remove `visibility: hidden` from the overlay by setting `visibility: initial`, but only after `display: none` has already been set. On show, we restore the CSS-specified visibility and display before setting `.is-open` to transition visibility, opacity and position.

**Pull request merged**